File: src/Parameters/Output/OutputParams.cpp
#include "OutputParams.h"
#include "../Vars/Vector.h"

OutputParams::OutputParams() {}

OutputParams::~OutputParams() {
    delete x_;
    delete alfa;
}

std::string *OutputParams::toString() {
    char number[15];
    std::string *rez = new std::string("Output Parameters:\n");

    sprintf(number, "\tk: %d;\n", k);
    (*rez) += number;

    (*rez) += "\tx_:\n\t\t";
    (*rez) += *(x_->toString());

    sprintf(number, "\tf_x_: %0.10f;\n", f_x_);
    (*rez) += number;

    (*rez) += "\talfa:\n\t\t";
    (*rez) += *(alfa->toString());

    sprintf(number, "\talfa_h: %0.10f;\n", alfa_h);
    (*rez) += number;

    return rez;
}

OutputParams *OutputParams::clone() {
    OutputParams *output = new OutputParams();
    output->k = k;
    output->x_ = x_ != nullptr ? x_->clone() : x_;
    output->f_x_ = f_x_;
    output->alfa = alfa != nullptr ? alfa->clone() : alfa;
    output->alfa_h = alfa_h;
    return output;
}

File: Parametrs/OutputParams.cs
using System;
using MethodsOptimization.src.Parametrs.Vars;
using System.Collections.Generic;

namespace MethodsOptimization.src.Parametrs
{
    class OutputParams: ICloneable
    {
        public OutputParams() { }

        /// <summary>
        /// Количество итераций
        /// </summary>
        public int K { get; set; } = 0;

        /// <summary>
        /// Интервал минимизации шага
        /// </summary>
        public Vector Alfa { get; set; }

        /// <summary>
        /// Шаг изменения
        /// </summary>
        public double Alfa_h { get; set; }

        /// <summary>
        /// К строке
        /// </summary>
        /// <returns></returns>
        new public string ToString()
        {
            string str = "Output:\n";
            str += "\tK: " + K + ";\n";
            str += "\tAlfa:\n\t" + Alfa.ToString().Replace("\t", "\t\t");
            str += "\tAlfa_h: " + Alfa_h + "\n";
            return str;
        }

        /// <summary>
        /// Клонировать
        /// </summary>
        /// <returns></returns>
        public object Clone()
        {
            OutputParams clone = new OutputParams();
            clone.K = K;
            clone.Alfa = (Vector)Alfa.Clone();
            clone.Alfa_h = Alfa_h;
            return clone;
        }
    }
}
